Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -4944,12 +4944,28 @@ static VisiblePositionRange constrainRangeToSelection(const VisiblePositionRange
         }
     }
 
-    auto makeString = [] (const VisiblePosition& start, const VisiblePosition& end) -> AttributedString {
+    auto isTextObscured = [] (const VisiblePosition& visiblePosition) -> bool {
+        if (RefPtr textControl = enclosingTextFormControl(visiblePosition.deepEquivalent())) {
+            if (is<HTMLInputElement>(textControl)) {
```

Comment:
```suggestion
            if (auto *input = dynamicDowncast<HTMLInputElement(textControl); input.input.isAutoFilledAndObscured()) {
            return true
            }
```

I think using `dynamicDowncast` might be a bit cleaner

Reply:
`auto* ` rather than `auto *` from Richard's comment, since it's a C++ object.